pupil-spa/src/app/services/speech/speech.service.ts

```diff
@@ -25,13 +48,32 @@ export class SpeechService {
    * @param utterance
    */
   speak(utterance: string): void {
+    this.cancel();
```

Comment:
is cancelled needed at the beginning? what happens if you remove it?

Reply:
cancel() ensures that any speech in progress will be curtailed immediately.  It's a preventative measure.